File: vortexdbg-api/src/main/java/com/vortexdbg/hook/Arm64HookContext.java
package com.vortexdbg.hook;

import com.vortexdbg.arm.context.EditableArm64RegisterContext;
import com.vortexdbg.pointer.VortexdbgPointer;
import com.sun.jna.Pointer;

import java.util.Stack;

class Arm64HookContext extends HookContext implements EditableArm64RegisterContext {

    private final EditableArm64RegisterContext registerContext;

    Arm64HookContext(Stack<Object> stack, EditableArm64RegisterContext registerContext) {
        super(stack);
        this.registerContext = registerContext;
    }

    @Override
    public void setXLong(int index, long value) {
        registerContext.setXLong(index, value);
    }

    @Override
    public void setStackPointer(Pointer sp) {
        registerContext.setStackPointer(sp);
    }

    @Override
    public long getXLong(int index) {
        return registerContext.getXLong(index);
    }

    @Override
    public int getXInt(int index) {
        return registerContext.getXInt(index);
    }

    @Override
    public VortexdbgPointer getXPointer(int index) {
        return registerContext.getXPointer(index);
    }

    @Override
    public long getFp() {
        return registerContext.getFp();
    }

    @Override
    public VortexdbgPointer getFpPointer() {
        return registerContext.getFpPointer();
    }

    @Override
    public int getIntArg(int index) {
        return registerContext.getIntArg(index);
    }

    @Override
    public long getLongArg(int index) {
        return registerContext.getLongArg(index);
    }

    @Override
    public VortexdbgPointer getPointerArg(int index) {
        return registerContext.getPointerArg(index);
    }

    @Override
    public long getLR() {
        return registerContext.getLR();
    }

    @Override
    public VortexdbgPointer getLRPointer() {
        return registerContext.getLRPointer();
    }

    @Override
    public VortexdbgPointer getPCPointer() {
        return registerContext.getPCPointer();
    }

    @Override
    public VortexdbgPointer getStackPointer() {
        return registerContext.getStackPointer();
    }

    @Override
    public int getIntByReg(int regId) {
        return registerContext.getIntByReg(regId);
    }

    @Override
    public long getLongByReg(int regId) {
        return registerContext.getLongByReg(regId);
    }
}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/hook/Arm64HookContext.kt
package com.vortexdbg.hook

import com.vortexdbg.arm.context.EditableArm64RegisterContext
import com.vortexdbg.pointer.VortexdbgPointer
import com.sun.jna.Pointer
import java.util.Stack

internal class Arm64HookContext(
    stack: Stack<Any?>,
    private val registerContext: EditableArm64RegisterContext
) : HookContext(stack), EditableArm64RegisterContext {

    override fun setXLong(index: Int, value: Long) {
        registerContext.setXLong(index, value)
    }

    override fun setStackPointer(sp: Pointer) {
        registerContext.setStackPointer(sp)
    }

    override fun getXLong(index: Int): Long {
        return registerContext.getXLong(index)
    }

    override fun getXInt(index: Int): Int {
        return registerContext.getXInt(index)
    }

    override fun getXPointer(index: Int): VortexdbgPointer {
        return registerContext.getXPointer(index)
    }

    override fun getFp(): Long {
        return registerContext.getFp()
    }

    override fun getFpPointer(): VortexdbgPointer {
        return registerContext.getFpPointer()
    }

    override fun getIntArg(index: Int): Int {
        return registerContext.getIntArg(index)
    }

    override fun getLongArg(index: Int): Long {
        return registerContext.getLongArg(index)
    }

    override fun getPointerArg(index: Int): VortexdbgPointer? {
        return registerContext.getPointerArg(index)
    }

    override fun getLR(): Long {
        return registerContext.getLR()
    }

    override fun getLRPointer(): VortexdbgPointer {
        return registerContext.getLRPointer()
    }

    override fun getPCPointer(): VortexdbgPointer {
        return registerContext.getPCPointer()
    }

    override fun getStackPointer(): VortexdbgPointer {
        return registerContext.getStackPointer()
    }

    override fun getIntByReg(regId: Int): Int {
        return registerContext.getIntByReg(regId)
    }

    override fun getLongByReg(regId: Int): Long {
        return registerContext.getLongByReg(regId)
    }
}
